.travis.yml

```diff
@@ -0,0 +1,11 @@
+language: node_js
+node_js:
+  - "0.12"
+before_script:
+  - 'npm install'
+  - 'bower install'
+  - 'npm install protractor'
+  - 'node_modules/protractor/bin/webdriver-manager update'
+  - 'webdriver-manager update'
+  - 'npm install grunt'
+  - 'npm install grunt-cli'
```

Comment:
Too much in your before script - instead of installing global installs you can run the file inside node_modules, e.g. ``` ./node_modules/.bin/bower install ```.